### ISSUE TITLE
Embedding layer need to support Adam opimizer update, How to?

### ISSUE THREAD

**Issue by typhoonzero** [PaddlePaddle/Paddle#7019]

**Comment by typhoonzero** [comment]
Basic functors seems needed in paddle for "lookup" based networks, see: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/docs_src/api_guides/python/state_ops.md#sparse-variable-updates

Also Adam optimizer can also use sparse gradient update like: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/adam.py#L174
